src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -25,6 +25,8 @@ internal record Parameter(string Name, FormattableString? Description, CSharpTyp
         public CSharpAttribute[] Attributes { get; init; } = Array.Empty<CSharpAttribute>();
         public bool IsOptionalInSignature => DefaultValue != null;
 
+        public Parameter WithRef(bool isRef = true) => this with { IsRef = isRef };
```

Comment:
I think maybe we could check if isRef is the same as the input, if so we could return `this` as we did in `WithAsync` in method signature